src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/util/DateTimeHelper.kt

```diff
@@ -5,12 +5,21 @@ import org.joda.time.DateTimeZone
 import org.joda.time.format.DateTimeFormat
 import java.time.LocalDate
 import java.time.LocalDateTime
+import java.time.OffsetDateTime
 import java.time.ZoneOffset
 import java.time.format.DateTimeFormatter
+import java.util.Locale
 
 class DateTimeHelper {
 
   companion object Helper {
+
+    fun oaSysUtcDateTimeFormatCorrecter(input: String): String {
```

Comment:
It is! Corrected!

Reply:
On second thoughts, using convertUtcDateTimeStringToIso8601Date breaks the e2e tests!
I think the pattern in it's formatter is subtly different. oaSysUtcDateTimeFormatCorrecter takes a DatetTimeStamp with millis  (which the OAsys Assessment dateCompleted has).

Reply:
I've modified the happy path test fixture to cover this.

Reply:
Can't find any CRNs on Preprod where that is happening, but if that is true, it makes me think that maybe there are other date fields where this could happen. In which case, maybe we remove the 'convertUtcDateTimeStringToIso8601Date' (or just move your code for oaSysUtcDateTimeFormatCorrecter into this method) to be on the safe side?